### PR TITLE
v10.7.0 - CSS file outputs

### PR DESCRIPTION
**Pull request by jamieomaguire**
v10.7.0
------------------------------
*October 10, 2022*

### Added
- CSS outputted files for typography, utilities and reset
- Snapshot tests for outputted files

### Changed
- Rename `compileToCss.js` to `cssCompiler.js`
- Rename `compileToCss` function to `compile`
- Path updates in snapshot test
- Replace multiline comments with single line so that they are removed by Sass without minification/compression
- Replace `prepare` with `prepack` as yarn lifecycle script to prepare `dist`

### Example of dist output:

Screenshot 2022-10-10 at 10 26 19

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,13 +77,26 @@
...
-    "prepare": "concurrently -n \"lint,compile,test\" -c \"blue,yellow,green\" \"yarn lint\" \"yarn compile\" \"yarn test\" --kill-others-on-fail",
+    "prepare": "concurrently -n \"lint,compile,test,prepare:css\" -c \"blue,yellow,green,#008080\" \"yarn lint\" \"yarn compile\" \"yarn test\" \"yarn prepare:css\" --kill-others-on-fail",
     "test": "concurrently -n \"test:js,test:scss\" -c \"cyan,magenta\" \"yarn test:js\" \"yarn test:scss\"",
     "test:js": "jest --config=jest.config.js",
     "test:scss": "jest --config=src/test/scss/jest.config.js",
     "test:build": "sass --no-source-map --load-path=node_modules --style=compressed src/scss:dist/css",
     "test:cover": "jest --collect-coverage",
-    "test:cover:CI": "cat coverage/lcov.info | coveralls"
+    "test:cover:CI": "cat coverage/lcov.info | coveralls",
+    "build:typography": "sass --load-path=node_modules/ src/scss/fozzie-typography.scss:dist/css/fozzie-typography.css --no-source-map",
```
Comment (jamieomaguire) [inline package.json:87]: if we decide to output more CSS files in the future then we can leverage the `cssCompiler.js` logic to do this programmatically

File: src/scss/base/_print.scss [hunk 2]
```diff
@@ -32,9 +32,9 @@
             content: ' (' attr(title) ')';
         }
 
-        /*
-        * Don't show links for images, or javascript/internal links
-        */
+        //
```
Comment (ashleynolan) [inline src/scss/base/_print.scss:35]: And these ones

File: src/scss/base/_print.scss [hunk 3]
```diff
@@ -48,7 +48,7 @@
         }
 
         thead {
-            display: table-header-group; /* h5bp.com/t */
+            display: table-header-group; // h5bp.com/t
```
Comment (ashleynolan) [inline src/scss/base/_print.scss:51]: And this one

File: src/scss/base/_reset.scss [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-    /**
-    * Fozzie Reset Styles
-    * =================================
-    * We use normalize.css for the bulk of our 'reset'
-    * but anything else 'reset-y' lives in here
-    *
-    * This is an optional component within Fozzie.
-    * If you'd like to use it in your project you can include it by adding `@include reset();` into your SCSS dependencies file.
-    */
+    // **
```
Comment (ashleynolan) [inline src/scss/base/_reset.scss:2]: This is ok to be loud

File: src/scss/base/_typography.scss [hunk 5]
```diff
@@ -5,30 +5,26 @@
...
-    *
-    * Documentation:
-    * TBC
-    */
-
-
-    /**
-    * Typography preloading styles
-    */
+    // **
```
Comment (ashleynolan) [inline src/scss/base/_typography.scss:8]: Any of these inside a mixin are fine to be loud comments as they'll be output in context in the CSS

**Review by ashleynolan — COMMENTED** [review]
Few comments on the comments – I haven't left comments on them all, but any comments that are next to classnames or inside a mixin are ok to be loud, as they'll be output in the correct context (and removed when minified later)

File: package.json [hunk 6]
```diff
@@ -74,13 +79,26 @@
...
     "lint:js": "eslint --ext .js .",
-    "prepare": "concurrently -n \"lint,compile,test\" -c \"blue,yellow,green\" \"yarn lint\" \"yarn compile\" \"yarn test\" --kill-others-on-fail",
+    "prepare": "concurrently -n \"lint,compile,test,prepare:css\" -c \"blue,yellow,green,#008080\" \"yarn lint\" \"yarn compile\" \"yarn test\" \"yarn prepare:css\" --kill-others-on-fail",
     "test": "concurrently -n \"test:js,test:scss\" -c \"cyan,magenta\" \"yarn test:js\" \"yarn test:scss\"",
     "test:js": "jest --config=jest.config.js",
     "test:scss": "jest --config=src/test/scss/jest.config.js",
     "test:build": "sass --no-source-map --load-path=node_modules --style=compressed src/scss:dist/css",
     "test:cover": "jest --collect-coverage",
-    "test:cover:CI": "cat coverage/lcov.info | coveralls"
+    "test:cover:CI": "cat coverage/lcov.info | coveralls",
```
Comment (ashleynolan) [inline package.json:88]: NIT – I'd put these into alphabetical order in the scripts list. So move autoprefixer and build to the top, prepare:css next to prepare etc.

File: src/scss/base/_grid.scss [hunk 7]
```diff
@@ -202,11 +195,11 @@
...
-    /**
-    * Gutter calcs – applied to grid columns in our grid (direct descendants only)
-    * Default: pixels (can look at changing to percentage)
-    * Usage: gutterCalc() or gutterCalc(false)
-    */
+    // **
+    // Gutter calcs – applied to grid columns in our grid (direct descendants only)
+    // Default: pixels (can look at changing to percentage)
+    // Usage: gutterCalc() or gutterCalc(false)
+    // **
```
[the remark this answers is not in the document]
Comment (jamieomaguire) [reply src/scss/base/_grid.scss:202]: Happy to revert, but does it make sense for a comment instructing sass usage to be included in the compiled CSS?

**Review by siggerzz — APPROVED** [review]

**Review by siggerzz — APPROVED** [review]

**Review by siggerzz — APPROVED** [review]

**Review by ashleynolan — APPROVED** [review]
All looks great!

One thing that would be great to add (in separate PR) would be a section in the README explaining the tests that run on the SCSS/CSS side and how contributors will need to update snapshots etc.

**Pull request merged**